### PR TITLE
Feature #685 progress bar in search dialog

### PR DESCRIPTION
**Pull request by tonyk24**

### CODE REVIEW

**Comment by donho** [comment]
Could you provid the info about this PR?

**Comment by tonyk24** [comment]
There is a progress bar at the bottom of the find dialog, and showing the remaining time of the search/count/replace operation.

The message queue is processed inbetween every  100th found occurance, making the find dialog non-blocking. All buttons but the close button is disabled during find. Pressing close cancels the operation and closes the dialog.

**Comment by tonyk24** [comment]
screenshot_progress2

**Comment by Ede123** [comment]
Is this only used for "Search" and "Replace" ("Highlight"?) tabs?
Or is "Search in Files" also updated?

I'm asking because there is already a progress bar for find in files and some consistency might be desirable.

File: PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp [hunk 1]
```diff
@@ -703,6 +703,8 @@ INT_PTR CALLBACK FindReplaceDlg::run_dlgProc(UINT message, WPARAM wParam, LPARAM
 			 _findClosePos.left = p.x;
 			 _findClosePos.top = p.y + 10;
 
+			 ::SendMessage(GetDlgItem(_hSelf, IDC_PROGRESSBAR), PBM_SETPOS, (WPARAM)0, 0);
```
Comment (MAPJe71) [inline PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp]: Please use C++ style typecast (see also [Guidelines for pull requests](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/CONTRIBUTING.md#guidelines-for-pull-requests)).

**Comment by tonyk24** [comment]
Currently show progress for each individual file while doing Find in Files. Perhaps not really consistant.

I see some options:
- Remove progress for individual files while doing Find in Files.
- Remove Find in files progress and just use progressbar in Find dialog
- Keep both but add the filename to status bar showing progress of individual file

**Comment by tonyk24** [comment]
Find/Replace/Mark: Will show remaining time, and pressing close in find dialog will cancel the operation. As a note, the replace operation can be cancelled mid-replace, without reversing the allready replaced items.
progress_find

Find in files: Will update the status bar find-dialog, but not possible to canel mid-file. Find in files is otherwise working as before.
progress_findinfiles

**Comment by donho** [comment]
I don't like the idea for putting more elements into find dialog which is already overloaded.

**Pull request closed without merging**